src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -49,7 +48,7 @@
 /**
  * Field Mapper for KNN vector type.
  */
-public class KNNVectorFieldMapper extends FieldMapper implements ArrayValueMapperParser {
```

Comment:
Why can we remove this?

Reply:
ArrayValueMapperParser is removed in ES version 7.8. So refactored around that